querydsl-sql-codegen/src/main/java/com/querydsl/sql/codegen/MetaDataExporter.java

```diff
@@ -374,7 +376,15 @@ private void handleTable(DatabaseMetaData md, ResultSet tables) throws SQLExcept
                 Map<String,InverseForeignKeyData> inverseForeignKeyData = keyDataFactory
                         .getExportedKeys(md, catalog, schema, tableName);
                 if (!inverseForeignKeyData.isEmpty()) {
-                    classModel.getData().put(InverseForeignKeyData.class, inverseForeignKeyData.values());
+                    Collection<InverseForeignKeyData> values = Collections2.filter(inverseForeignKeyData.values(), new Predicate<InverseForeignKeyData>() {
```

Comment:
filtering style should be consistent with namingStrategy.shouldGenerateForeignKey